users/src/test/java/io/crate/operation/user/UserPriviligeValidationTest.java

```diff
@@ -0,0 +1,200 @@
+package io.crate.operation.user;
+
+import com.google.common.collect.ImmutableSet;
+import io.crate.exceptions.PermissionDeniedException;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.TableIdent;
+import io.crate.metadata.doc.DocSchemaInfoFactory;
+import io.crate.metadata.table.Operation;
+import io.crate.metadata.table.SchemaInfo;
+import io.crate.metadata.table.TableInfo;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import org.elasticsearch.common.settings.Settings;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class UserPriviligeValidationTest extends CrateDummyClusterServiceUnitTest {
```

Comment:
Wouldn't it be possible to use the SqlExecutor / TableDefinitions? Those are "real" tables (except that they're out-of-sync with the clusterState)